package.json

```diff
@@ -8,21 +8,24 @@
         "url": "https://github.com/gosling-lang/gosling.js"
     },
     "homepage": "https://gosling-lang.github.io/gosling.js/",
-    "main": "dist/gosling.js",
+    "main": "dist/gosling.es.js",
     "module": "dist/gosling.es.js",
     "types": "dist/src/index.d.ts",
     "files": [
         "dist"
     ],
+    "type": "module",
```

Comment:
My understanding of this is that with the exception of some configs, we want all files to be treated as ESM so we use this option.

Reply:
Yup! Making things work nice (like Vite) might require upgrading those deps.

Reply:
That makes sense! I did try upgrading Vite before, but it we're using a plugin ([@vitejs/plugin-react-refresh](https://www.npmjs.com/package/@vitejs/plugin-react-refresh)) which requires this version of vite. However, I didn't realize until now that the package is now deprecated in favor of [@vitejs/plugin-react](https://www.npmjs.com/package/@vitejs/plugin-react). 

I'll switch to @vitejs/plugin-react and upgrade vite if that sounds good!